python/hsfs/engine/python.py

```diff
@@ -963,36 +963,40 @@ def acked(err, msg):
             producer.flush()
             progress_bar.close()
 
-        # start backfilling job
-        # if topic didn't exist, always run the backfill job to reset the offsets except if it's a multi insert
+        # start materialization job
+        # if topic didn't exist, always run the materialization job to reset the offsets except if it's a multi insert
         if (
             not isinstance(feature_group, ExternalFeatureGroup)
             and reset_offsets
             and not feature_group._multi_part_insert
         ):
-            if offline_write_options is not None and not offline_write_options.get(
-                "start_offline_backfill", True
+            if offline_write_options is not None and not (
+                offline_write_options.get("start_offline_materialization", True)
+                or offline_write_options.get("start_offline_backfill", True)
             ):
                 warnings.warn(
-                    "This is the first ingestion after an upgrade or backup/restore, running backfill job even though `start_offline_backfill` was set to `False`.",
+                    "This is the first ingestion after an upgrade or backup/restore, running materialization job even though `start_offline_materialization` was set to `False`.",
                     util.FeatureGroupWarning,
                 )
-            feature_group.backfill_job.run(
-                args=feature_group.backfill_job.config.get("defaultArgs", "")
+            feature_group.materialization_job.run(
```

Comment:
Isn't this misaligned? Looking at the code (assuming the if statement above is fixed), if the user doesn't want to start the job, we only print a warning, but then we keep going and we actually run the job nevertheless?

Or am I missign somthing?

Reply:
hmm, checking master and indentation is the same there. But the warning saying `running materialization job even though `start_offline_materialization` was set to `False`` I guess that was intended?

Reply:
Ok I see what you mean but then the indentation is correct.